services/drupal/web/modules/custom/epa_workflow/src/Plugin/Action/EPABulkRepublishAction.php

```diff
@@ -0,0 +1,211 @@
+   * @param $plugin_definition
+   * @param EntityTypeManagerInterface $entity_type_manager
+   * @param AccountInterface $account
+   * @param ModerationInformationInterface $moderation_info
+   * @param $validator
+   */
+  public function __construct(
+    array            $configuration,
+                     $plugin_id,
+                     $plugin_definition,
+    EntityTypeManagerInterface $entity_type_manager,
+    AccountInterface $account,
+    ModerationInformationInterface $moderation_info,
+    StateTransitionValidationInterface $validator
+  )
+  {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entity_type_manager;
+    $this->account = $account;
+    $this->moderationInfo = $moderation_info;
+    $this->validator = $validator;
+  }
+
+  /**
+   * @param ContainerInterface $container
+   * @param array $configuration
+   * @param $plugin_id
+   * @param $plugin_definition
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition)
+  {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('current_user'),
+      $container->get('content_moderation.moderation_information'),
+      $container->get('content_moderation.state_transition_validation')
+    );
+  }
+
+  /**
+   * @param $entity
+   * @return \Drupal\Core\StringTranslation\TranslatableMarkup
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function execute($entity = NULL)
+  {
+    // Ensure we're acting on the default revision
+    if (!$entity->isDefaultRevision()) {
+      $entity = \Drupal::entityTypeManager()->getStorage($entity->getEntityTypeId())->load($entity->id());
+    }
+      $entity->set('moderation_state', 'published');
+      if ($entity instanceof RevisionLogInterface) {
+        $entity->setRevisionLogMessage('Bulk re-publishing.');
+        $entity->setRevisionUserId($this->account->id());
+        $entity->setRevisionCreationTime(\Drupal::time()
+          ->getRequestTime());
+      }
+      $entity->save();
+      return $this->t('Content has been re-published.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access($object,  AccountInterface $account = NULL, $return_as_object = FALSE) {
+    if (!$object->isDefaultRevision()) {
+      $object = \Drupal::entityTypeManager()->getStorage($object->getEntityTypeId())->load($object->id());
+    }
+    if (!$object || !$object instanceof ContentEntityInterface) {
+      $result = AccessResult::forbidden('Not a valid entity.');
+      return $return_as_object ? $result : $result->isAllowed();
+    }
+    $object = $this->loadLatestRevision($object);
```

Comment:
I'm pretty sure this is a mistake, and is causing the bug found here: https://forumone.atlassian.net/browse/EPAD8-1942?focusedCommentId=511987 .  The whole point of this new actionis that we want to be acting on the default/current revision, not the latest revision.